### ISSUE TITLE
error NG1010: Expected array when reading the NgModule.declarations of ExtendedModule

### ISSUE THREAD

**Issue by silentsnooc** [angular/angular#35388]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [x] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

Yes, the previous version in which this bug was not present was: < 9.0.0

### Description

I just upgraded to Angular 9 but as I am compiling the code I am getting the error below. To me, this looks like an issue with Flex Layout but we've been told this would be an issue with Angular (see https://github.com/angular/flex-layout/issues/1196).

## 🔥 Exception or Error
<pre><code>
ERROR in Failed to compile entry-point @angular/flex-layout/extended (module as esm5) due to compilation errors:
node_modules/@angular/flex-layout/esm5/extended.es5.js:903:35 - error NG1010: Expected array when reading the NgModule.declarations of ExtendedModule

903                     declarations: __spreadArrays(ALL_DIRECTIVES),
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
</code></pre>


## 🌍 Your Environment
<pre><code>
Angular CLI: 9.0.1
Node: 10.15.2
OS: linux x64

Angular: 9.0.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, material, material-moment-adapter
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.1
@angular-devkit/build-angular     0.900.1
@angular-devkit/build-optimizer   0.900.1
@angular-devkit/build-webpack     0.900.1
@angular-devkit/core              9.0.1
@angular-devkit/schematics        9.0.1
@angular/cli                      9.0.1
@angular/flex-layout              9.0.0-beta.29
@angular/http                     7.2.16
@ngtools/webpack                  9.0.1
@schematics/angular               9.0.1
@schematics/update                0.900.1
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2

</code></pre>

**Anything else relevant?**

The content of my package.json 

```json
{
  "name": "mahlzeit-admin",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "node server.js",
    "build": "ng build --prod",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "bundle-report": "ng build --prod --aot --stats-json && webpack-bundle-analyzer dist/stats.json"
  },
  "engines": {
    "node": "8.12.0",
    "npm": "6.4.1"
  },
  "private": true,
  "dependencies": {
    "@agm/core": "^1.0.0-beta.3",
    "@angular/animations": "9.0.0",
    "@angular/cdk": "^9.0.0",
    "@angular/cli": "9.0.1",
    "@angular/common": "9.0.0",
    "@angular/compiler": "9.0.0",
    "@angular/compiler-cli": "9.0.0",
    "@angular/core": "9.0.0",
    "@angular/flex-layout": "^9.0.0-beta.29",
    "@angular/forms": "9.0.0",
    "@angular/http": "7.2.16",
    "@angular/material": "^9.0.0",
    "@angular/material-moment-adapter": "^9.0.0",
    "@angular/platform-browser": "9.0.0",
    "@angular/platform-browser-dynamic": "9.0.0",
    "@angular/router": "9.0.0",
    "@fortawesome/angular-fontawesome": "^0.6.0",
    "@fortawesome/fontawesome-svg-core": "^1.2.4",
    "@fortawesome/free-brands-svg-icons": "^5.3.1",
    "@fortawesome/free-solid-svg-icons": "^5.3.1",
    "@ngx-share/button": "^7.0.0",
    "@ngx-share/core": "^7.0.0",
    "@types/quill": "^2.0.2",
    "@types/stripe-v3": "^3.1.6",
    "@zxing/ngx-scanner": "^3.0.0",
    "angular-mentions": "^1.1.3",
    "chart.js": "^2.7.2",
    "core-js": "^3.6.4",
    "express": "^4.16.3",
    "fuse.js": "^3.4.4",
    "http-status-codes": "^1.3.0",
    "mahlzeit-rest-client": "0.0.1",
    "moment": "^2.24.0",
    "ng-pick-datetime": "^7.0.0",
    "ngx-logger": "^4.1.2",
    "ngx-moment": "^3.1.0",
    "ngx-quill": "^8.0.0",
    "npm-check-updates": "^4.0.1",
    "paypal-checkout": "^4.0.239",
    "quill": "^1.3.6",
    "quill-mention": "^2.1.2",
    "rxjs": "^6.5.2",
    "typescript": "3.7.5",
    "zone.js": "^0.10.2"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^0.900.1",
    "@angular/language-service": "9.0.0",
    "@types/googlemaps": "^3.30.16",
    "@types/jasmine": "^3.3.0",
    "@types/jasminewd2": "^2.0.6",
    "@types/node": "~13.7.1",
    "codelyzer": "^5.2.1",
    "enhanced-resolve": "^4.1.0",
    "express-http-proxy": "^1.5.0",
    "jasmine-core": "~3.5.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.4.1",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage-istanbul-reporter": "^2.0.4",
    "karma-jasmine": "~3.1.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "protractor": "^5.4.1",
    "ts-node": "~8.6.2",
    "tslint": "^6.0.0",
    "webpack-bundle-analyzer": "^3.0.3"
  }
}
```

**Comment by gkalpak** [comment]
I believe this should be fixed by #35191. Can you please try using the build artifacts from [here][1] (see [there][2] for instructions) and confirm that is makes the issue go away?

[1]: https://circleci.com/gh/angular/angular/621158#artifacts/containers/0
[2]: https://github.com/angular/angular/blob/eef047bc5/docs/DEVELOPER.md#getting-packages-from-build-artifacts

**Comment by KingDarBoja** [comment]
@gkalpak Using the provided steps does solve the issue for `aws-amplify-angular` package.

**Comment by silentsnooc** [comment]
@gkalpak It turns out I messed up some things when upgrading to Angular 9. I had to fix my angular.json etc. in order to make it work again. This was on me. Thanks for helping!

**Comment by hoang-innomizetech** [comment]
@silentsnooc What did you do to fix this issue? Can you share it?

**Comment by silentsnooc** [comment]
@hoang-innomizetech I actually just created a new Angular 9 project and used a diff tool to compare the config files (angular.json, tsconfig etc) and made the according changes.

**Comment by bobbydowling** [comment]
Would you mind being more specific? :)

**Comment by ShahoG** [comment]
I'm also having this issue with the aws-amplify-angular package, please advise

**Comment by KingDarBoja** [comment]
@ShahoG Copy and paste on your package.json:

```json
"@angular/compiler-cli": "https://621158-24195339-gh.circle-artifacts.com/0/angular/compiler-cli-pr35191-b93c5372e9.tgz"
```

Run npm install and that should fix the issue.

**Comment by gkalpak** [comment]
Keep in mind that [using PR build artifacts][1] (as suggested by @KingDarBoja above) is only guaranteed to work temporarily (i.e. it is not guaranteed that the artifacts will remain available  long-term). It is fine as a very temporary solution, mainly to verify that a specific PR fixes an issue with your app.

[1]: https://github.com/angular/angular/blob/e084835fb/docs/DEVELOPER.md#getting-packages-from-build-artifacts

**Comment by KingDarBoja** [comment]
@gkalpak Did the PR got merged already and released into latest Angular version? I saw it got closed without merge but as far as I know, this is how the Angular PRs work.

**Comment by gkalpak** [comment]
@KingDarBoja, yes, the PR has been merged into master and the 9.0.x branch and will be included in the upcoming releases (probably later this week).
